var/spack/repos/builtin/packages/extrae/package.py

```diff
@@ -35,17 +35,23 @@ class Extrae(Package):
        OpenMP, CUDA, OpenCL, pthread, OmpSs"""
     homepage = "https://tools.bsc.es/extrae"
     url      = "https://ftp.tools.bsc.es/extrae/extrae-3.4.1-src.tar.bz2"
+    version('3.7.1', sha256='95810b057f95e91bfc89813eb8bd320dfe40614fc8e98c63d95c5101c56dd213', preferred=True)
     version('3.4.1', '69001f5cfac46e445d61eeb567bc8844')
 
     depends_on("mpi")
-    depends_on("dyninst")
     depends_on("libunwind")
     depends_on("boost")
     depends_on("libdwarf")
     depends_on("papi")
     depends_on("elf", type="link")
     depends_on("libxml2")
 
+    variant('dyninst', default=False, description="Use dyninst for dynamic code installation")
+    depends_on('dyninst@:9.9.99', type=('build', 'link'), when='+dyninst')
```

Comment:
No need to specify the type, this is the default. Also, you can do `dyninst@:9` to make things simpler.